scio-bigquery/src/main/scala/com/spotify/scio/bigquery/BigQueryIO.scala

```diff
@@ -247,6 +263,35 @@ object BigQueryTypedTable {
     ): WriteParam = apply(s, wd, cd, td, tp, DefaultExtendedErrorInfo)(defaultInsertErrorTransform)
   }
 
+  private[this] def tableRow(table: Table): BigQueryTypedTable[TableRow] =
+    BigQueryTypedTable(
+      beam.BigQueryIO.readTableRows(),
+      beam.BigQueryIO.writeTableRows(),
+      table,
+      BigQueryUtils.convertGenericRecordToTableRow(_, _)
+    )
+
+  private[this] def genericRecord(table: Table): BigQueryTypedTable[GenericRecord] =
+    BigQueryTypedTable(
+      _.getRecord(),
+      identity[GenericRecord],
+      (genericRecord: GenericRecord, _: TableSchema) => genericRecord,
+      table
+    )
+
+  /**
+   * Creates a new instance of [[BigQueryTypedTable]] based on the supplied [[Format]].
+   *
+   * NOTE: LogicalType support when using `Format.GenericRecord` has some caveats:
+   * Reading: Bigquery types DATE, TIME, DATIME will be read as STRING
+   * Writting: Supports LogicalTypes only for DATE and TIME.
+   *           DATETIME is not yet supported. https://issuetracker.google.com/issues/140681683
+   */
+  def apply(table: Table, format: Format): BigQueryTypedTable[format.F] = format match {
+    case Format.GenericRecord => genericRecord(table).asInstanceOf[BigQueryTypedTable[format.F]]
+    case Format.TableRow      => tableRow(table).asInstanceOf[BigQueryTypedTable[format.F]]
+  }
```

Comment:
Why not simply expose `def genericRecord(table: Table)` and `def tableRow(table: Table)` and skip the `Format` trait entirely ?
```suggestion
def genericRecord(table: Table): BigQueryTypedTable[GenericRecord] = ???
def tableRow(table: Table): BigQueryTypedTable[TableRow] = ???
```

Reply:
Yeah, on a second thought is not really adding anything... I liked the idea initially of switching format based on param. Ok let's keep this simple

Reply:
Oh wait i remembered why i kept this (not just because i liked it lol)... because of the overloaded methods in `ScioContextSyntax`


```scala
def bigQueryTable(table: Table): SCollection[TableRow]

def bigQueryTable(table: Table, format: Format): SCollection[format.F]
```

Reply:
Bin compat in Scio is IMHO an impossible goal bc. Beam or its dependencies happily break it with every release.

Reply:
Yeah it makes sense for `ScioContextSyntax`